pkg/kopialib/storage/storage_interface.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/pkg/errors"
+)
+
+type StorageType string
+
+const (
+	TypeS3        StorageType = "S3"
+	TypeAzure     StorageType = "Azure"
+	TypeFileStore StorageType = "FileStore"
+	TypeGCP       StorageType = "GCP"
+)
+
+type Storage interface {
+	Connect() (blob.Storage, error)
+	SetOptions(context.Context, map[string]string) error
+	WithCreate(bool)
```

Comment:
I think these two methods should not be part of the interface. They can be part of the actual implementation though. 
Because they are more related to the impl. of the interface, whether we want to create the impl with options with created etc.

Reply:
I think SetOptions should be part of interface because we cannot initialize a storage without options. I can remove Create method from implementation